app/models/shop.rb

```diff
@@ -24,4 +29,13 @@ def self.last
   def self.all
     DB[:shops].all
   end
+
+  def self.find(id)
+    h = DB[:shops].first(:id => id) #sequel doc uses first for retrieving record
+    Shop.new(h) #returns a shop object with hash argument
+  end
+
+  def set(id, name)
```

Comment:
maybe we can call it `update` and no parameter is needed here.

if you create a method reader for `id`, then we can expect a behavior like this:

```ruby
s = Shop.find(1) # let's suppose it has name: "Tea Shop"
s.name = "Updated Shop"
s.update
```